src/goSuggest.ts

```diff
@@ -199,7 +199,17 @@ export class GoCompletionItemProvider implements vscode.CompletionItemProvider {
 										paramSnippets.push('${' + (i + 1) + ':' + param + '}');
 									}
 								}
-								item.insertText = new vscode.SnippetString(suggest.name + '(' + paramSnippets.join(', ') + ')');
+								let currentWordIndex = lineText.toLowerCase().lastIndexOf(currentWord.toLowerCase());
+								let nextSymbolIndex = currentWordIndex + currentWord.length;
+								let newSnippetString = null;
+								// Avoid adding snippet for function suggest when cursor is followed by ()
+								// i.e: met() -> method()()
+								if ((nextSymbolIndex < lineText.length) && (lineText.charAt(nextSymbolIndex) === '(')) {
+									newSnippetString = new vscode.SnippetString(suggest.name);
+								} else {
```

Comment:
Why not skip setting the `item.insertText` altogether if the cursor is followed by `()`

Reply:
Because keeping the text insert will still autocomplete the function name sans the param snippets